python/pyspark/sql/connect/dataframe.py

```diff
@@ -255,10 +255,21 @@ def distinct(self) -> "DataFrame":
         )
 
     def drop(self, *cols: "ColumnOrString") -> "DataFrame":
```

Comment:
This is an interesting case where one could argue for implementing the behavior on the client side instead of the server.

Reply:
IIUC,  there will be two RPC if we implement it on the client side
1, `all_cols = self.columns` to fetch the schema;
2, build the plan

with a dedicated proto mesage, we only need one RPC.

Reply:
I think we have sort of building the consensus that we prefer re-using the proto than ask clients do duplicate work.